### PR TITLE
1107 bug fix bool operator for comparison formula

### LINKED ISSUE #1107: [🐞 Bug]: Fix bool operator for comparison formula
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://pypi.org/project/blue-prints/) of Blueprints.

### Issue description

\_\_bool\_\_ should use _comparison_operator

### Reproducible example

```python

```

### Expected behavior

_No response_

### Installed versions

```

```

### PR DESCRIPTION
**Pull request by SZeltaat**
## Description

Hi Enrique,
This is the fix (Sina edition). My PR text is not as fancy as yours, but my code is human written and checked. 😄 

The unity check for formula 5-7 and 5-8 was wrong. When you have lhs > rhs, you should calculate the u.c. = rhs / lhs. 

Fixes #1107 
## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected how comparison-based formulas determine satisfaction (including proper handling for negative values and real boolean results).
  * Updated unity-check calculations for sway imperfections and slenderness-related checks to match the intended expressions.
  * Improved aggregated results output so LaTeX clearly renders **“OK”** vs **“Not OK”** consistently.
* **Tests**
  * Expanded regression coverage for comparison truthiness and aggregated LaTeX rendering.
  * Updated expected values to reflect corrected unity-check and comparison behavior.

### CODE REVIEW

File: blueprints/codes/formula.py [hunk 1]
```diff
@@ -211,29 +211,29 @@ def unity_check(self) -> float:
...
             print("Condition satisfied")
 
         Returns
         -------
         bool
-            True if unity_check <= 1.0 (condition is satisfied), False otherwise.
+            True if the comparison condition is satisfied, False otherwise.
         """
-        return self.unity_check <= 1.0
+        return self._comparison_operator()(self.lhs, self.rhs)
```
Comment (egarciamendez) [inline blueprints/codes/formula.py]: One thing worth adding: wrap this in `bool()`.

If `_evaluate_lhs` returns a numpy float, `operator.le` gives back a `numpy.bool`, and Python requires `__bool__` to return a real `bool`:

```
operator.le(np.float64(1.0), 2.0)  ->  numpy.bool

without bool() cast : TypeError: __bool__ should return bool, returned numpy.bool
with    bool() cast : True
```

This is not hypothetical: it is what `Form8Dot41InclinationCompressionField` hit before the `float()` wrappers went into `math_helpers`, and what (8.60) hit for me. Those wrappers cover it as long as everyone goes through `cot`, `sec` and `csc`, but a formula that reaches for `np.sqrt` or `np.tan` directly in `_evaluate_lhs` brings it straight back.

`AggregatedComparisonFormula.__bool__` does not need it, since `all()` and `any()` already return a real bool.

**Review by egarciamendez — CHANGES_REQUESTED** [review]

File: blueprints/codes/formula.py (hunk 1 above)
Comment (SZeltaat) [reply blueprints/codes/formula.py]: Good point! I could not really think of a situation that bool would be necessary.
Made the changes and added tests for this edge case

**Review by egarciamendez — APPROVED** [review]
